tests/unit/disco/components/TestAddon.js

```diff
@@ -443,42 +444,102 @@ describe(__filename, () => {
   describe('addon with type static theme', () => {
     const fullImage = 'https://addons.cdn.mozilla.net/full/54321.png';
     const newAddonName = 'Summertime';
+    let install = sinon.spy();
+    let enable = sinon.spy();
 
-    it('renders the correct image', () => {
-      const shallowRoot = renderAddon({
+    const renderWithStaticTheme = (props = {}) => {
+      return renderAddon({
         addon: {
           ...result,
           name: newAddonName,
+          type: ADDON_TYPE_STATIC_THEME,
           previews: [
             {
               ...fakePreview,
               image_url: fullImage,
             },
           ],
-          type: ADDON_TYPE_STATIC_THEME,
         },
+        status: UNINSTALLED,
+        install,
+        enable,
+        hasAddonManager: true,
+        getAddonEnabledStatus: sinon.stub().returns(false),
+        ...props,
       });
-      const image = shallowRoot.find('.Addon-theme-header-image');
+    };
+
+    it('renders the correct image', () => {
+      const root = renderWithStaticTheme();
+      const image = root.find('.Addon-theme-header-image');
       expect(image).toHaveProp('src', fullImage);
       expect(image).toHaveProp('alt', `Preview of ${newAddonName}`);
     });
+
+    it("calls install and enable helper functions when clicking on the static theme's header image if hasAddonManager is true", async () => {
+      install = sinon.spy();
+      enable = sinon.spy();
+
+      const root = renderWithStaticTheme();
+      const themeImage = root.find('.theme-image-link');
+      await themeImage.simulate('click', createFakeEvent());
+      await sinon.assert.calledOnce(install);
+      sinon.assert.calledOnce(enable);
+    });
+
+    it("does not call enable helper functions when clicking on the static theme's header image if addon is already enabled", async () => {
+      install = sinon.spy();
+      enable = sinon.spy();
+      const root = renderWithStaticTheme({
+        getAddonEnabledStatus: sinon.stub().returns(true),
+      });
+      const themeImage = root.find('.theme-image-link');
+      await themeImage.simulate('click', createFakeEvent());
+      await sinon.assert.called(install);
```

Comment:
that's so weird... Maybe we should not use `simulate('click')` and call `await root.instance().installStaticTheme()` directly.. (cc @kumar303)

Reply:
this works, thank you.... the way @kumar303 suggested below works here too. is one way better than the other?

Reply:
it is fine to use Kumar's proposal (that is: getting the prop and invoking it)

Reply:
ok sure I'll update this.. thanks!